### PR TITLE
feat(default_ad_api): add a node to show guidance messages depending on the system status

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add guide message node using ADAPI.

## Tests performed

```bash
$ ros2 run default_ad_api guide.py 
The vehicle pose is not estimated. Plase set an initial pose or check GNSS.
The route is not set. Please set a goal pose.
The topic rate error is detected. Please check [control,planning] components.
The vehicle is ready. Change the operation mode to autonomous.
The vehicle is driving autonomously.
The vehicle has reached the goal of the route. Please reset a route.
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/default_ad_api/script/guide.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            print(message)
+            self.previous_message = message
+
+    def check_localization(self):
+        if self.localization.msg is None:
+            return "The localization state is not received. Please check ADAPI and localization component works correctly."
+        elif self.localization.msg.state == LocalizationInitializationState.INITIALIZED:
+            return self.check_routing()
+        else:
+            return "The vehicle pose is not estimated. Plase set an initial pose or check GNSS."
```
Comment (kenji-miyake) [inline system/default_ad_api/script/guide.py:87]:
```suggestion
            return "The vehicle pose is not estimated. Please set an initial pose or check GNSS."
```

**Review by kenji-miyake — COMMENTED** [review]
Is this a node for debugging? How about writing a brief usage in README?

**Comment by isamu-takagi** [comment]
> Is this a node for debugging? How about writing a brief usage in README?

@kenji-miyake Thank you. I updated readme.

File: system/default_ad_api/README.md [hunk 2]
```diff
@@ -9,3 +11,22 @@ This package is a default implementation AD API.
...
+This is a sample to call API using HTTP.
+
+## Guide message script
+
+This is a debug script to check the conditions for transition to autonomous mode.
+
+```bash
+$ ros2 run default_ad_api guide.py
+
+The vehicle pose is not estimated. Plase set an initial pose or check GNSS.
```
Comment (kenji-miyake) [inline system/default_ad_api/README.md:26]:
```suggestion
The vehicle pose is not estimated. Please set an initial pose or check GNSS.
```

**Review by kenji-miyake — APPROVED** [review]
Regarding the title, I personally think something like this is more descriptive.
`feat(default_ad_api): add a node to show guidance messages depending on the system initialization status`

I'll leave it to you.

**Pull request merged**